src/Webapi/Webapi__Dom/Webapi__Dom__NodeList.re

```diff
@@ -2,7 +2,7 @@ type t = Dom.nodeList;
 
 [@bs.val] external toArray : t => array(Dom.node) = "Array.prototype.slice.call";
 
-[@bs.send.pipe : t] external forEach : ((Dom.node, int) => unit) => unit = "";
+[@bs.send.pipe : t] external forEach : ((Dom.element, int) => unit) => unit = "";
```

Comment:
Hmm, why did you do this? It's a NodeList after all, and while some functions like `querySelectorAll` will return only elements, `Node.childNodes` will not.

Reply:
[MDN documentation](https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach) only talk about elements.

And TypeScript have
```
NodeListOf<Element>.forEach(callbackfn: (value: Element, key: number, parent: NodeListOf<Element>) => void, thisArg?: any): void
```

Have you a better source for the type of this value ?

Reply:
As I said, `Node.childNodes` returns a `NodeList` of all kinds of nodes. MDN is a wiki, you shouldn't trust it 100%. Look at the standards documents instead, like https://dom.spec.whatwg.org/#nodelist.

TypeScript seems to have parametetized it. What you're looking at is the type signature specifically for a `NodeList` of `Element`s. You could probably also have something like a `NodeListOf<Node>`, in which case the type of `forEach` would be

```
NodeListOf<Node>.forEach(callbackfn: (value: Node, key: number, parent: NodeListOf<Node>) => void, thisArg?: any): void
```

I'm sure we could do the same, but that's a bigger change. Give it a shot if you want.